pkg/storaged/iscsi-panel.jsx

```diff
@@ -0,0 +1,295 @@
+                       dialog.open({ Title: _("Change iSCSI Initiator Name"),
+                                     Fields: [
+                                         { TextInput: "name",
+                                           Title: _("Name"),
+                                           Value: name
+                                         }
+                                     ],
+                                     Action: {
+                                         Title: _("Change"),
+                                         action: function (vals) {
+                                             return client.manager_iscsi.call('SetInitiatorName',
+                                                                              [ vals.name,
+                                                                                { }
+                                                                              ]);
+                                         }
+                                     }
+                       });
+                   });
+        }
+
+        function cmp_session(path_a, path_b) {
+            var a = client.iscsi_sessions[path_a];
+            var b = client.iscsi_sessions[path_b];
+            var a_name = a.data["target_name"] || "";
+            var b_name = b.data["target_name"] || "";
+
+            return a_name.localeCompare(b_name);
+        }
+
+        function make_session(path) {
+            var session = client.iscsi_sessions[path];
+
+            function iscsi_remove() {
+                self.setState({ armed: false });
+                return session.Logout({ 'node.startup': { t: 's', v: "manual" } });
+            }
+
+            var actions = null;
+            if (self.state.armed)
+                actions = (
+                    <StorageButton kind="danger" onClick={iscsi_remove}>
+                        <span className="pficon pficon-delete"/>
+                    </StorageButton>
+                );
+
+            return (
+                <OverviewSidePanelRow client={client}
+                                      kind="array"
+                                      name={session.data["target_name"] || ""}
+                                      detail={session.data["persistent_address"] + ":" +
+                                              session.data["persistent_port"]}
+                                      actions={actions}/>
+            );
+        }
+
+        const toggle_armed = (event) => {
+            if (!event || event.button !== 0)
+                return;
+            this.setState({ armed: !this.state.armed });
+        }
+
+        var sessions = Object.keys(client.iscsi_sessions).sort(cmp_session).map(make_session);
+
+        var actions = [
+            sessions.length > 0 ?
+            <button className={"btn btn-default fa fa-check" + (this.state.armed? " active" : "")}
+                    onClick={toggle_armed}/> : null,
+            "\n",
```

Comment:
These line break texts amidst the actions looks odd. Is that just "avoid CSS" laziness, or is there something functional behind it?

Reply:
I guess it is "CSS laziness".  With the old mustache code, we had newlines between the elements, which render as small spaces between the buttons.  This emulates this behavior with React.

I'll try to get this done with CSS.

Reply:
Not urgent, fine to do in a separate PR if you prefer (or not at all). I was just wondering where that came from.

Reply:
Skipped.